cli/tests/test_diff_command.rs

```diff
@@ -1736,12 +1736,209 @@ fn test_diff_skipped_context() {
+       1    1: a
+       2    2: b
+       3    3: c
+       4    4: d
+       5    5: eE
+       6    6: f
+       7    7: g
+       8    8: h
+       9    9: i
+      10   10: j
+    === 1 line at end
+    Modified regular file file1:
+       1    1: a
+       2    2: b
+       3    3: c
+       4    4: d
+       5    5: e
+       6    6: fF
+       7    7: g
+       8    8: h
+       9    9: i
+      10   10: j
+    "#);
+}
+
+#[test]
+fn test_diff_skipped_context_0_fromsettings() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    test_env.add_config(
+        r#"
+[diff]
+context = 0
+        "#,
+    );
+
+    std::fs::write(repo_path.join("file1"), "a\nb\nc\nd").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["describe", "-m", "=== Left side of diffs"]);
+
+    test_env.jj_cmd_ok(&repo_path, &["new", "@", "-m", "=== Must skip 2 lines"]);
+    std::fs::write(repo_path.join("file1"), "A\nb\nc\nD").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "@-", "-m", "=== Don't skip 1 line"]);
+    std::fs::write(repo_path.join("file1"), "A\nb\nC\nd").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "@-", "-m", "=== No gap to skip"]);
+    std::fs::write(repo_path.join("file1"), "a\nB\nC\nd").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "@-", "-m", "=== 1 line at start"]);
+    std::fs::write(repo_path.join("file1"), "a\nB\nc\nd").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["new", "@-", "-m", "=== 1 line at end"]);
+    std::fs::write(repo_path.join("file1"), "a\nb\nC\nd").unwrap();
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &["log", "-Tdescription", "-p", "--no-graph", "--reversed"],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    === Left side of diffs
+    Added regular file file1:
+            1: a
+            2: b
+            3: c
+            4: d
+    === Must skip 2 lines
+    Modified regular file file1:
+       1    1: aA
+        ...
+       4    4: dD
+    === Don't skip 1 line
+    Modified regular file file1:
+       1    1: aA
+       2    2: b
+       3    3: cC
+       4    4: d
+    === No gap to skip
+    Modified regular file file1:
+       1    1: a
+       2    2: bB
+       3    3: cC
+       4    4: d
+    === 1 line at start
+    Modified regular file file1:
+       1    1: a
+       2    2: bB
+        ...
+    === 1 line at end
+    Modified regular file file1:
+        ...
+       3    3: cC
+       4    4: d
+    "###);
+}
+
 #[test]
 fn test_diff_skipped_context_nondefault() {
     let test_env = TestEnvironment::default();
     test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
     let repo_path = test_env.env_root().join("repo");
 
+    // Verify that the command line overrides a user setting.
+    test_env.add_config(
+        r#"
+[diff]
+context = 4
+        "#,
+    );
```

Comment:
nit: the default config includes `diff.context = 3`, so we don't have to override it?